compiler/rustc_ast_passes/src/feature_gate.rs

```diff
@@ -14,6 +14,17 @@ use rustc_span::Span;
 use tracing::debug;
 
 macro_rules! gate_feature_fn {
+    ($visitor: expr, $has_feature: expr, $span: expr, $name: expr, $explain: expr, $help: expr) => {{
+        let (visitor, has_feature, span, name, explain, help) =
+            (&*$visitor, $has_feature, $span, $name, $explain, $help);
+        let has_feature: bool = has_feature(visitor.features);
+        debug!("gate_feature(feature = {:?}, span = {:?}); has? {}", name, span, has_feature);
+        if !has_feature && !span.allows_unstable($name) {
+            feature_err_issue(&visitor.sess.parse_sess, name, span, GateIssue::Language, explain)
+                .span_help(span, help)
```

Comment:
Because we have a single span, I think this should be a bare `help` call (without a span) so the output isn't too verbose/redundant. I would _not_ close #76011, as that is asking for a structured suggestion, which we might be able to provide in the future (but this current change is a great mitigation).